doc/modules/classification_threshold.rst

```diff
@@ -0,0 +1,168 @@
+categorical prediction (class label). Scores are obtained from :term:`predict_proba` or
+:term:`decision_function`. The former returns posterior probability estimates for each
+class, while the latter returns a decision score for each class. The decision score is a
+measure of how strongly the sample is predicted to belong to the positive class (e.g.,
+the distance to the decision boundary). A decision rule is then defined by thresholding
+the scores, leading to a class label for each sample. Those labels are obtained with
+:term:`predict`.
+
+For binary classification in scikit-learn, class labels are obtained by associating the
+positive class with posterior probability estimates greater than 0.5 (obtained with
+:term:`predict_proba`) or decision scores greater than 0 (obtained with
+:term:`decision_function`).
+
+Here, we show an example that illustrates the relation between posterior
+probability estimates and class labels::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.tree import DecisionTreeClassifier
+    >>> X, y = make_classification(random_state=0)
+    >>> classifier = DecisionTreeClassifier(max_depth=2, random_state=0).fit(X, y)
+    >>> classifier.predict_proba(X[:4])
+    array([[0.94   , 0.06   ],
+           [0.94   , 0.06   ],
+           [0.04..., 0.95...],
+           [0.04..., 0.95...]])
+    >>> classifier.predict(X[:4])
+    array([0, 0, 1, 1])
+
+While these approaches are reasonable as default behaviors, they are not ideal for
+all cases. The context and nature of the use case defines the expected behavior of the
+classifier and thus, the strategy to convert soft predictions into hard predictions. We
+illustrate this point with an example.
+
+Let's consider a scenario where a predictive model is being deployed to assist medical
+doctors in detecting tumors. In this setting, doctors will be most likely interested in
+correctly identifying all patients with cancer so that they can provide them with the
+right treatment. In other words, doctors prioritize achieving a high recall rate,
+meaning they want to identify all cases of cancer without missing any patients who have
+it. This emphasis on recall comes, of course, with the trade-off of potentially more
+false-positive predictions, reducing the precision of the model, but that is a risk
+doctors are willing to take. Consequently, when it comes to deciding whether to classify
+a patient as having cancer or not, it may be more beneficial to classify them as
+positive for cancer when the posterior probability estimate is lower than 0.5.
+
+Post-tuning the decision threshold
+==================================
+
+One solution to address the problem stated in the introduction is to tune the decision
+threshold of the classifier once the model has been trained. The
+:class:`~sklearn.model_selection.TunedThresholdClassifier` tunes this threshold using an
+internal cross-validation. The optimum threshold is chosen to maximize a given metric
```

Comment:
Today, I would indeed consider it as a design and methodological flaw. I don't know what level of justification do you think we should add-on here:

```
... tunes this threshold using an internal cross-validation since the scikit-learn
:term:`predict` API does not offer this flexibility.
```

Reply:
> we still don't have an API for path-like algorithms and efficient CV

@amueller What do you mean by that?

> is this tool just a workaround for a design flaw wherein a parameter to BaseClassifier is not exposed

and
> Today, I would indeed consider it as a design and methodological flaw.

@glemaitre @jnothman Shouldn't we fix this design flaw? Do you have suggestions?

Reply:
> Shouldn't we fix this design flaw? Do you have suggestions?

This is something that I would introduce in 2.0 because we will surely break some API. In this case, we could the freedom to think about a good API.

Reply:
I guess what I am missing is a roadmap or a plan/vision.